flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedTaskManagerTrackerTest.java

```diff
@@ -319,14 +435,118 @@ public void testGetStatistics() {
                 TASK_EXECUTOR_CONNECTION.getInstanceID(),
                 defaultSlotResource,
                 SlotState.ALLOCATED);
-        taskManagerTracker.addPendingTaskManager(
-                new PendingTaskManager(ResourceProfile.fromResources(4, 200), 1));
 
-        assertThat(taskManagerTracker.getFreeResource(), is(ResourceProfile.fromResources(6, 700)));
-        assertThat(taskManagerTracker.getRegisteredResource(), is(totalResource));
-        assertThat(taskManagerTracker.getNumberRegisteredSlots(), is(10));
-        assertThat(taskManagerTracker.getNumberFreeSlots(), is(8));
+        PendingTaskManager pendingTaskManager =
+                new PendingTaskManager(ResourceProfile.fromResources(4, 200), 1);
+        taskManagerTracker.allocateTaskManagersAccordingTo(
+                new ResourceAllocationResult.Builder()
+                        .addPendingTaskManagerAllocate(pendingTaskManager)
+                        .addAllocationOnPendingResource(
+                                jobId,
+                                pendingTaskManager.getPendingTaskManagerId(),
+                                ResourceProfile.fromResources(4, 200))
+                        .build());
+
+        assertThat(taskManagerTracker.getFreeResource())
+                .isEqualTo(ResourceProfile.fromResources(6, 700));
+        assertThat(taskManagerTracker.getRegisteredResource()).isEqualTo(totalResource);
+        assertThat(taskManagerTracker.getNumberRegisteredSlots()).isEqualTo(10);
+        assertThat(taskManagerTracker.getNumberFreeSlots()).isEqualTo(8);
+        assertThat(taskManagerTracker.getPendingResource())
+                .isEqualTo(ResourceProfile.fromResources(4, 200));
+    }
+
+    @Test
+    void testTimeoutForUnusedTaskManager() throws Exception {
```

Comment:
It seems that `testTimeoutForUnusedTaskManager` in `FineGrainedSlotManagerTest` is not needed.